bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,873 @@
+    }
+
+    /**
+     * Adapter for HTTP 2.0 session status events.
+     *
+     * The session must be permanently connected, so it ignores onIdleTimeout() events.
+     * It also handles the following fatal events by notifying the owner.
+     *
+     * <li>onClose()</li>
+     * <li>onFailure()</li>
+     * <li>onGoAway()</li>
+     * <li>onReset()</li>
+     *
+     * @author Andrew Fiddian-Green - Initial Contribution
+     */
+    private static class SessionAdapter extends Session.Listener.Adapter implements AdapterErrorHandler {
+
+        private final Clip2Bridge owner;
+
+        protected SessionAdapter(Clip2Bridge owner) {
+            this.owner = owner;
+        }
+
+        @Override
+        public void fatalError(Error error) {
+            owner.fatalError(this, error);
+        }
+
+        @Override
+        public void onClose(@Nullable Session session, @Nullable GoAwayFrame frame) {
+            fatalError(Error.CLOSED);
+        }
+
+        @Override
+        public void onFailure(@Nullable Session session, @Nullable Throwable failure) {
+            fatalError(Error.FAILURE);
+        }
+
+        @Override
+        public void onGoAway(@Nullable Session session, @Nullable GoAwayFrame frame) {
+            fatalError(Error.GOAWAY);
+        }
+
+        @Override
+        public boolean onIdleTimeout(@Nullable Session session) {
+            return false;
+        }
+
+        @Override
+        public void onReset(@Nullable Session session, @Nullable ResetFrame frame) {
+            fatalError(Error.RESET);
+        }
+    }
+
+    private static final String APPLICATION_ID = "org-openhab-binding-hue-clip2";
+    private static final String APPLICATION_KEY = "hue-application-key";
+
+    private static final String FORMAT_URL_CONFIG = "http://%s/api/0/config";
+    private static final String FORMAT_URL_RESOURCE = "https://%s/clip/v2/resource/";
+    private static final String FORMAT_URL_REGISTER = "http://%s/api";
+    private static final String FORMAT_URL_EVENTS = "https://%s/eventstream/clip/v2";
+
+    private static final int CLIP2_MINIMUM_VERSION = 1948086000;
+    private static final int TIMEOUT_SECONDS = 10;
+    private static final int CHECK_ALIVE_SECONDS = 300;
+
+    private static final ResourceReference BRIDGE = new ResourceReference().setType(ResourceType.BRIDGE);
+
+    /**
+     * Static method to attempt to connect to a Hue Bridge, get its software version, and check if it is high enough to
+     * support the CLIP 2 API.
+     *
+     * @param hostName the bridge IP address.
+     * @return returns without any exception if the bridge is online and it does support CLIP 2.
+     * @throws ApiException if was not possible to connect or another error was encountered.
+     * @throws IllegalArgumentException if the hostName is bad.
+     * @throws IllegalStateException if it was possible to connect but CLIP 2 is not supported.
+     */
+    public static void testSupportsClip2(String hostName)
```

Comment:
The method `public static void testSupportsClip2(String hostName)` is a STATIC method that is called before the binding would be allowed to create a thing instance. If the test fails, then no Things are created and therefore no property would be available.

```
     * Static method to attempt to connect to a Hue Bridge, get its software version, and check if it is high enough to
     * support the CLIP 2 API.
```

Reply:
That's a valid argument for discovery, but currently `Clip2BridgeHandler` will also call this method on each initialization. That seems to be more than needed? Firmware downgrade is highly unusual, so the only probable case would be invalid manual configuration. Perhaps a failure later during initialization could be picked up and used rather than this additional call?

Reply:
I will keep this method call, but a) rename it, and b) refine it according to your other suggestions.